### PR TITLE
Fix calling Load with None returns bad error

### LINKED ISSUE #19190: Calling load with None argument gives unhelpful error
When calling Load from the Python interface with a `None` argument for the Filename:
```python
Load(Filename=None)
```
Gives the following error - we should explicitly tell the user they are passing a `None` value instead.
```
ValueError: Problem when setting Filename. This is the detailed error description: When converting 
parameter "Filename": TypeRegistry::retrieve(): No PropertyValueHandler registered for type 'class 
std::vector<class std::vector<class std::basic_string<char,struct std::char_traits<char>,class 
std::allocator<char> >,class std::allocator<class std::basic_string<char,struct std::char_traits<char>,class 
std::allocator<char> > > >,class std::allocator<class std::vector<class std::basic_string<char,struct 
std::char_traits<char>,class std::allocator<char> >,class std::allocator<class std::basic_string<char,struct 
std::char_traits<char>,class std::allocator<char> > > > > >'
```

### PR DESCRIPTION
**Pull request by DavidFair**
Description of work.
This PR fixes an issue with `Load` being called in Python with `Filename=None` passed. Previously it would print a long error message describing the Python container. Now it prints a valid error saying that a `None` is not acceptable.

Additionally I tided `simpleapi` to fix the PEP8 white space issues (which should reduce the number of warnings in flake8) ~~and remove `import os` from the `simpleapi` which was unused into the various scripts which relied on this implicit import.~~ this was too much work so in the next maintenance cycle I'll open another PR to finish this properly. 

**To test:**
Ensure all builds pass
Run the following and check it gives a sane error message
`Load(Filename=None)`

Fixes #19190 .

*Does not need to be in the release notes. - Only noticed internally*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [x] Is the code of an acceptable quality?
- [x] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [x] Do changes function as described? Add comments below that describe the tests performed?
- [x] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?

- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by gvardany** [comment]
Loading none throws the expected error. But should you not remove in this PR the `import os` from the 5 algorithms touched then?

**Comment by DavidFair** [comment]
I'm in two minds about this - technically I should do them all in a separate PR which I will do in the future and normally I would completely agree with you. 
However equally this makes the scripts correct without affecting them. Reverting the changes - then putting them back again in a later PR seems wasteful to me.

**Comment by gvardany** [comment]
OK :shipit: then

File: Framework/PythonInterface/mantid/simpleapi.py [hunk 1]
```diff
@@ -806,68 +850,72 @@ def ignore_property(name, ignore_regex):
...
-
+                raise RuntimeError('Internal error. Unknown property type encountered. "%s" '
+                                   'on algorithm "%s" is not understood by '
+                                   'Python. Please contact development team' % (name, algm_obj.name()))
     nvals = len(retvals)
     nlhs = lhs[0]
-    if nlhs > 1 and nvals != nlhs:
+
+    # If we have more than one value but not the same number of values throw
+    if 1 < nlhs != nvals:
```
Comment (martyngigg) [inline Framework/PythonInterface/mantid/simpleapi.py]: Is this correct? It doesn't read very well to me as it takes too long to think and remember the operator precedence rules to see if it is correct? Was this a suggestion from the flake8 warning or somewhere else?

Comment (DavidFair) [inline Framework/PythonInterface/mantid/simpleapi.py]: Chaining expressions is something Python encourages you to do so my IDE warns me on this line. I'm not sure whether it is a flake warning but it tends to be flake warnings are a subset of the warnings I get locally. 

How about tidying it up to both remove the warning and make the intention a bit clearer:
Original:
```python
    if nlhs > 1 and nvals != nlhs:
        # There is a discrepancy in the number are unpacking variables
        # Let's not have the more cryptic unpacking error raised
        raise RuntimeError("%s is trying to return %d output(s) but you have provided %d variable(s). "
                           "These numbers must match." % (func_name, nvals, nlhs))
    if nvals > 0:
        ret_type=namedtuple(func_name+"_returns",retvals.keys())
        ret_value=ret_type(**retvals)
        if nvals==1:
            return ret_value[0]
        else:
            return ret_value
    else:
        return None
```
Into something like:
```python
if nlhs == 0:
    return None

if nvals != nlhs:
   if nlhs == 1:
        pass   # This is fine as we do not need to unpack
   else:
               # There is a discrepancy in the number are unpacking variables
        # Let's not have the more cryptic unpacking error raised
        raise RuntimeError("%s is trying to return %d output(s) but you have provided %d variable(s). "
                           "These numbers must match." % (func_name, nvals, nlhs))

# Unpack and return
ret_type=namedtuple(func_name+"_returns",retvals.keys())
ret_value=ret_type(**retvals)
return ret_value[0] if nvals == 1 else ret_value

Comment (martyngigg) [inline Framework/PythonInterface/mantid/simpleapi.py]: I'll have to live with disagreeing with Python over `1 < nlhs != nvals` then I think but to be honest it just looks weird to me.

:+1: For your new version. This reads better and makes the intention of the code much clearer.

**Comment by peterfpeterson** [comment]
I changed the ship to a squirrel so it doesn't accidentally match a filter for things that are ready to be merged.

**Comment by DavidFair** [comment]
It turns out there is a subtle bug in the original implementation of unpacking returned values from a function simple API executes. 
If there is a function called `myFunc` which returns 3 values in Python we should return a tuple of three values in the following case
`foo=myFunc(...)`
Currently we return just the first result.

I'm going to revert commits  `39a24b1` and `f7cab6b` and cherry pick them into their own PR as they will require some additional work.

File: Framework/PythonInterface/mantid/simpleapi.py (hunk 1 above)
Comment (martyngigg) [inline Framework/PythonInterface/mantid/simpleapi.py]: Can we put this back to what it was originally: `if nlhs > 1 and nvals != nlhs:` ?

File: Framework/PythonInterface/mantid/simpleapi.py [hunk 2]
```diff
@@ -806,68 +850,72 @@ def ignore_property(name, ignore_regex):
...
         # Let's not have the more cryptic unpacking error raised
         raise RuntimeError("%s is trying to return %d output(s) but you have provided %d variable(s). "
                            "These numbers must match." % (func_name, nvals, nlhs))
     if nvals > 0:
-        ret_type=namedtuple(func_name+"_returns",retvals.keys())
-        ret_value=ret_type(**retvals)
-        if nvals==1:
+        ret_type = namedtuple(func_name+"_returns", retvals.keys())
+        ret_value = ret_type(**retvals)
+        if nvals == 1:
```
Comment (martyngigg) [inline Framework/PythonInterface/mantid/simpleapi.py]: Also, I'm not sure I see the where the bug is? 

This checks if `nvals==1` which is if there is `1` value provided not if there is 1 value on the left hand side. I don't think the usage examples for [`GetEi`](http://docs.mantidproject.org/nightly/algorithms/GetEi-v2.html#usage) would work if this was broken.

File: Framework/PythonInterface/mantid/simpleapi.py (hunk 1 above)
Comment (DavidFair) [inline Framework/PythonInterface/mantid/simpleapi.py]: 👍

File: Framework/PythonInterface/mantid/simpleapi.py (hunk 2 above)
Comment (DavidFair) [inline Framework/PythonInterface/mantid/simpleapi.py]: Looking at master it seems my indentation went wrong which was throwing me off.

I definitely think this needs a clean up considering the confusion that has arisen from it but I agree its for a separate PR so I'll revert it back to the original form

**Comment by DavidFair** [comment]
Just a gentle nudge could we look at this PR again?

**Comment by martyngigg** [comment]
I'm happy with the this now. @gvardany Any other comments?

**Review by martyngigg — APPROVED** [review]

**Comment by gvardany** [comment]
Not really, please go ahead.

**Pull request merged**